packages/commonwealth/client/scripts/controllers/app/login.ts

```diff
@@ -178,15 +187,36 @@ export async function updateActiveAddresses({
   } else if (app.user.activeAccounts.length === 0) {
     // no addresses - preview the community
   } else {
-    const existingAddress = app.roles.getDefaultAddressInCommunity({
-      chain: chain.id,
-    });
+    // Find all addresses in the current community for this account, sorted by last used date/time
+    const communityAddressesSortedByLastUsed =
```

Comment:
As per Jake's comment, we should remove the changes to roles.ts. Then this becomes:
```
    const communityAddressesSortedByLastUsed = app.user.addresses
      .filter(a => a.community.id === chain.id)
      .sort((a, b) => b.lastActive.diff(a.lastActive));
```